tests/auth_tests/test_auth_backends.py

```diff
@@ -667,3 +683,30 @@ def test_backend_path_login_with_explicit_backends(self):
         user = User.objects.create_user(self.username, 'email', self.password)
         self.client._login(user, self.other_backend)
         self.assertBackendInSession(self.other_backend)
+
+
+@override_settings(
+    AUTHENTICATION_BACKENDS=['django.contrib.auth.backends.AllowInactiveUsersModelBackend'],
+)
+class AllowInactiveUsersModelBackendTest(TestCase):
+
+    def setUp(self):
+        self.user = User.objects.create_user('test', 'test@example.com', 'test')
+        self.user.is_active = False
+        self.user.save()
+
+    def test_authenticate(self):
+        self.assertEqual(authenticate(username='test', password='test'), self.user)
+
```

Comment:
@timgraham this test won't pass until #6124 is merged, because inactive user is rejected by test client. What would you suggest in this case? Thanks.